### PR TITLE
Remove all manual assigments of nrunner

### PR DESCRIPTION
**Pull request by clebergnu**
nrunner is the default since version 92.0.  With a number of bug fixes
since then (such as for job API usage) there should be no reason for
manual selections.

Also, let's start to phase out whenever possible the "N(ext) Runner"
name, since it's now actually the current runner.

Signed-off-by: Cleber Rosa <crosa@redhat.com>

### CODE REVIEW

File: examples/jobs/passjob_custom.py [hunk 1]
```diff
@@ -6,17 +6,15 @@
...
 # Custom method (no discovery, no guess, no magic)
 # Since there is no magic, we need to pass a suite name, otherwise a uuid4 will
 # be used for suite.name. Also resolver.references will be ignored (Avocado will not
 # creating tests suites for you).
 
-suite1 = TestSuite(config=config,
+suite1 = TestSuite(config={},
                    tests=[Runnable("noop", "noop")], name='suite1')
-suite2 = TestSuite(config=config,
+suite2 = TestSuite(config={},
```
Comment (beraldoleal) [inline examples/jobs/passjob_custom.py:16]: There is any special need to pass an empty config here? IIRC, config is optional for TestSuite and for Job.

File: examples/jobs/passjob_custom.py [hunk 2]
```diff
@@ -6,17 +6,15 @@
...
                    tests=[Runnable("noop", "noop")], name='suite2')
 
-with Job(config, [suite1, suite2]) as j:
+with Job({}, [suite1, suite2]) as j:
```
Comment (beraldoleal) [inline examples/jobs/passjob_custom.py:19]: And maybe would be cleaner with:

```python
with Job(test_suites=[suite1, suite2]) as j:
```

File: selftests/unit/test_job.py [hunk 3]
```diff
@@ -94,17 +94,15 @@ def test_job_test_suite_not_created(self):
         self.assertIsNone(self.job.test_suite)
 
     def test_suite_not_started(self):
-        suite = TestSuite('empty-suite', {'run.test_runner': 'nrunner'})
+        suite = TestSuite('empty-suite', {})
```
Comment (beraldoleal) [inline selftests/unit/test_job.py:97]: Same here.

File: examples/jobs/passjob_custom.py (hunk 1 above)
Comment (clebergnu) [reply examples/jobs/passjob_custom.py:16]: I was on the fence about hinting that users can pass a config (as this is an example) or not... I'm still on the fence :)

Comment (clebergnu) [reply examples/jobs/passjob_custom.py:16]: I guess looking at the other examples, I'm not on the fence anymore.  I'll remove them.

**Comment by clebergnu** [comment]
Closing in favor of #5093

**Pull request closed without merging**